Description: split dspam_notify script to have a separate configuration file
Author: Debian DSPAM Maintainers <pkg-dspam-misc@lists.alioth.debian.org>
Forwarded: not-needed

--- /dev/null
+++ b/src/tools/dspam_notify.conf
@@ -0,0 +1,29 @@
+#!/usr/bin/perl
+
+# Who will the notifications be sent from?
+$FROM_EMAIL = 'dspam@example.org';
+$FROM_NAME = 'DSPAM Filter';
+  
+# What will the notification subject be?
+$SUBJECT = 'Daily Spam Quarantine Summary';
+
+# What text to display in the body?
+$BODY = qq!<p>This report has been sent to you from the Anti-Spam service hosted at example.org. Below is a list of items in your quarantine area. You can view or release a message by clicking on the links (right). If you no longer wish to receive these reports then you may change the option on the 'Preferences' page.</p>!;
+
+# Quarantine URL
+$DSPAM_URL = 'https://dspam.example.org';
+
+# Maximum of entries to show in mail
+$MAX_ITEMS = 200;
+
+# Address of your SMTP server? localhost should be fine.
+$SERVER = 'localhost';
+
+# Port of your SMTP server? 25 should be fine
+$PORT = '25';
+
+# Enable User Preference Checking (Very CPU Intensive!!!) Not Recommended for more than 500 email accounts.
+$PREF_CHECK = 0;
+
+# Configuration was successful
+1;
--- a/src/tools/dspam_notify.in
+++ b/src/tools/dspam_notify.in
@@ -4,35 +4,15 @@
 
 use Net::SMTP;
 
-# Enter the location of you dspam.conf, dspam binary and path to dspam_admin/dspam_stats.
-$DSPAMCONF = '@sysconfdir@/dspam.conf';
-$DSPAM_BINARY = '@bindir@/@dspam_transformed@';
-$BINDIR = '@bindir@';
-
-# Who will the notifications be sent from?
-$FROM_EMAIL = 'dspam@example.org';
-$FROM_NAME = 'DSPAM Filter';
-  
-# What will the notification subject be?
-$SUBJECT = 'Daily Spam Quarantine Summary';
-
-# What text to display in the body?
-$BODY = qq!<p>This report has been sent to you from the Anti-Spam service hosted at example.org. Below is a list of items in your quarantine area. You can view or release a message by clicking on the links (right). If you no longer wish to receive these reports then you may change the option on the 'Preferences' page.</p>!;
-
-# Quarantine URL
-$DSPAM_URL = 'https://dspam.example.org';
+if (!(-e "/etc/dspam/dspam_notify.conf") || !(-r "/etc/dspam/dspam_notify.conf")) {
+  die("Missing configuration file: /etc/dspam/dspam_notify.conf");
+}
+require "/etc/dspam/dspam_notify.conf";

-# Maximum of entries to show in mail
-$MAX_ITEMS = 200;
-
-# Address of your SMTP server? localhost should be fine.
-$SERVER = 'localhost';
-
-# Port of your SMTP server? 25 should be fine
-$PORT = '25';
-
-# Enable User Preference Checking (Very CPU Intensive!!!) Not Recommended for more than 500 email accounts.
-$PREF_CHECK = 0;
+# Enter the location of you dspam.conf, dspam binary and path to dspam_admin/dspam_stats.
+$DSPAMCONF = '/etc/dspam/dspam.conf';
+$DSPAM_BINARY = '/usr/bin/dspam';
+$BINDIR = '/usr/bin';
 
 ######################################
 # No need to config below this point.#
